### ISSUE TITLE
Update from 1.5.5 to 1.5.9 broke LDAP plugin

### ISSUE THREAD

**Issue by jmechalas** [FOGProject/fogproject#445]
After updating from 1.5.5 to 1.5.9, our users could no longer login using LDAP. Error logs showed:

```
AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught Exception: Port is not valid ldap/ldaps port in /var/www/sse-fog/lib/plugins/ldap/class/ldap.class.php:147
```

In the source code, it looks like the plugin is trying to get a setting LDAP_PORTS, and is failing. It seems this should be in the `globalSettings` database table, but it was missing. I tried to use the LDAP Plugin Configuration page to set this value, but the form gave this error:

```
Settings Update Fail

Not all elements in filter or ports setting are integer
```

In the end, I had to add the LDAP_PORTS setting to the `globalSettings` table by hand, setting it to "389,636" (for those who are also running into this issue).

**Comment by jmechalas** [comment]
FOG Server 1.5.9
Ubuntu 18.04.4 LTS
Kernel 4.15.0-154-generic
Intel(R) Xeon(R) CPU E3-1280 v6 @ 3.90GHz

**Comment by jmechalas** [comment]
I also noticed while I was in there that the error output that gets logged to /var/log/apache/error.log includes the user's credentials. :(

**Comment by Sebastian-Roth** [comment]
@jmechalas Thanks for bringing this up. Unfortunately it's a known issue in FOG that plugins have an issue when there is a database schema update needed because this is done on plugin install only. So re-installing the plugin would have fixed the issue most probably.

Though I have to admit that I am not exactly sure there was an issue with the LDAP plugin in FOG 1.5.9 that we fixed in `dev-branch` already. Hmmm, looking at the [commit history](https://github.com/FOGProject/fogproject/commits/dev-branch/packages/web/lib/plugins/ldap) I don't think that's the case. So the above statement should be correct.

> I also noticed while I was in there that the error output that gets logged to /var/log/apache/error.log includes the user's credentials.

Can you please post an example error message here (removing actual credentials)?

**Comment by mechalas** [comment]
In the virtual host error log (error-fog.log) we see:

```
[Thu Nov 18 09:23:50.654438 2021] [proxy_fcgi:error] [pid 27864] [client 10.243.51.164:62606] AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught Exception: Port is not valid ldap/ldaps port in /var/www/sse-fog/lib/plugins/ldap/class/ldap.class.php:147\nStack trace:\n#0 /var/www/sse-fog/lib/plugins/ldap/class/ldap.class.php(256): LDAP->_ldapUp()\n#1 /var/www/sse-fog/lib/plugins/ldap/hooks/ldappluginhook.hook.php(126): LDAP->authLDAP('@@USERNAME@@', '@@PASSWORD@@')\n ....
```

**Comment by Sebastian-Roth** [comment]
@mechalas Thanks a lot for the information! Saved me some time for setting up things to test myself. Still took me a while to find the time to look at this, sorry.

This is caused by stacktrace printing all the information. From what I read about this we could either:
* use `set_exception_handler()` and `restore_exception_handler()` to silence the full stacktrace output
* switch to other means of error handling for missing port information in [plugins/ldap/class/ldap.class.php](https://github.com/FOGProject/fogproject/blob/master/packages/web/lib/plugins/ldap/class/ldap.class.php#L147)

@mastacontrola You know more about FOG's error handling than I do. What do you think?

**Comment by Sebastian-Roth** [comment]
@mechalas said:
> In the virtual host error log (error-fog.log) we see:
```
... ldappluginhook.hook.php(126): LDAP->authLDAP('@@USERNAME@@', '@@PASSWORD@@')\n ...
```
I just stumbled upon [php.ini settings](https://php.watch/versions/8.0/throwable-stacktrace-param-max-length) we could use to hide function parameters. Either we could restrict the length (I think it defaults to 15 characters) of completely hide the parameter printing.

@mastacontrola Do you think it would hurt to disable parameter printing for stack traces globally?

**Comment by Sebastian-Roth** [comment]
@mechalas Shoud be fixed now - see b0234bd.
